File: frontend/src/app/core/models/employee.ts
export type EmployeeStatus = 'active' | 'on_leave' | 'inactive';

export interface Employee {
  id: string;
  name: string;
  email: string;
  title: string;           // job title / role
  department_id: string;
  department?: { id: string; name: string };
  status: EmployeeStatus;
  start_date: string;
  created_at?: string;
  updated_at?: string;
}

/** Shape used by forms (no id, no timestamps) */
export interface EmployeeFormValue {
  name: string;
  email: string;
  title: string;
  department_id: string;
  status: EmployeeStatus;
  start_date: string;
}

File: backend/app/schemas/employee.py
from datetime import datetime, date
from decimal import Decimal
from pydantic import BaseModel, EmailStr
from app.schemas.common import BaseSchema


class DeptRef(BaseSchema):
    id: str
    name: str


class ManagerRef(BaseSchema):
    id: str
    name: str


class EmployeeCreate(BaseModel):
    name: str
    email: EmailStr
    title: str
    salary: Decimal | None = None
    status: str = "active"
    start_date: date
    department_id: str
    manager_id: str | None = None
    profile_picture_url: str | None = None


class EmployeeUpdate(BaseModel):
    name: str | None = None
    email: EmailStr | None = None
    title: str | None = None
    salary: Decimal | None = None
    status: str | None = None
    start_date: date | None = None
    department_id: str | None = None
    manager_id: str | None = None
    profile_picture_url: str | None = None


class EmployeeOut(BaseSchema):
    id: str
    name: str
    email: str
    title: str
    salary: Decimal | None = None
    status: str
    start_date: date
    profile_picture_url: str | None = None
    department_id: str
    department: DeptRef | None = None
    manager_id: str | None = None
    manager: ManagerRef | None = None
    created_at: datetime
    updated_at: datetime
